### ISSUE TITLE
SFR dataset 6d and 6e stored by itmp position instead of nseg. 

### ISSUE THREAD

**Issue by jlarsen-usgs** [modflowpy/flopy#546]
In ModflowSfr1.load() and ModflowSfr1.write_file() channel geometry (dataset 6d) and channel flow data (dataset 6e) are stored by their position relative to itmp instead of the corresponding nseg number defined in SFR dataset 6a.  Lines 832 - 863

```
                for j in range(itmp):
                    dataset_6a = _parse_6a(f.readline(), option)
                    current_aux[j] = dataset_6a[-1]
                    dataset_6a = dataset_6a[:-1]  # drop xyz
                    icalc = dataset_6a[1]
                    dataset_6b = _parse_6bc(f.readline(), icalc, nstrm,
                                            isfropt,
                                            reachinput, per=i)
                    dataset_6c = _parse_6bc(f.readline(), icalc, nstrm,
                                            isfropt,
                                            reachinput, per=i)

                    current[j] = dataset_6a + dataset_6b + dataset_6c

                    if icalc == 2:
                        # ATL: not sure exactly how isfropt logic functions for this
                        # dataset 6d description suggests that this line isn't read for isfropt > 1
                        # but description of icalc suggest that icalc=2 (8-point channel) can be used with any isfropt
                        if i == 0 or nstrm > 0 and not reachinput:  # or isfropt <= 1:
                            dataset_6d = []
                            for k in range(2):
                                dataset_6d.append(
                                    _get_dataset(f.readline(), [0.0] * 8))
                                # dataset_6d.append(list(map(float, f.readline().strip().split())))
                            current_6d[j + 1] = dataset_6d
                    if icalc == 4:
                        nstrpts = dataset_6a[5]
                        dataset_6e = []
                        for k in range(3):
                            dataset_6e.append(
                                _get_dataset(f.readline(), [0.0] * nstrpts))
                        current_6e[j + 1] = dataset_6e
```
where j is defined by 

According to the online documentation for SFR

_Note 21: Item 6 must be completed ITMP times. The data need not be defined in sequential order by stream segment number. All active segments in the stream network must be defined for each stress period through a combination of ITMP and Item 6 and NP and Item 7._
--

This creates issues for adding new segments and renumbering the SFR network. A solution would be to store the data by the segment number it is associated with from dataset 6a. This would allow for renumbering of the SFR network using stream elevation data or other user supplied schemes.

Secondly: nss is defined as:

```
@property
    def nss(self):
        # number of stream segments
        return len(np.atleast_1d(self.segment_data[0]))
```
which assumes all stream segments are active in the first stress period of a simulation. A solution to this issue could be...

```
@property
def nss(self):
    return len(set(self.reach_data.iseg))
```

**Comment by langevin-usgs** [comment]
@jlarsen-usgs is this fixed now?

**Comment by jlarsen-usgs** [comment]
This is fixed with PR #547
